### PR TITLE
doc: add socket console backend for virtio-console

### PR DESCRIPTION
**Pull request by gaojunhao0504**
add new console backend of socket for virtio-console

Signed-off-by: Gao Junhao <junhao.gao@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
@liuyuan-GitHub , Could you help to review this doc? Thanks!

**Comment by NanlinXie** [comment]
@JasonChenCJ , Could you help to review this doc? Thanks!

File: doc/developer-guides/hld/virtio-console.rst [hunk 1]
```diff
@@ -198,3 +201,31 @@ The File backend only supports console output to a file (no input).
...
 
       console=hvc0
+
+SOCKET
+======
+
+When virtio-console socket_type is appointed to server, device model
+will create a unix domain socket for server VM to bind and listen.
+Then other VM virtio-console socket_type is set to client, these
+VM can connect to the unix domain socket and communicate with server VM.
```
Comment (ywan170) [inline doc/developer-guides/hld/virtio-console.rst]: The virtio-console socket-type can be set as socket server or client. Device model will create an unix domain socket if appointed the socket_type as server, then server VM or another user VM can bind and listen for communication requirement. If appointed to client, please make sure the socket server is ready prior to launch device model.

File: doc/developer-guides/hld/virtio-console.rst [hunk 2]
```diff
@@ -198,3 +201,31 @@ The File backend only supports console output to a file (no input).
...
+
+#. Add a pci slot to the device model (``acrn-dm``) command line, adjusting
+   the ``</path/to/file.sock>`` to your use case in the VM2 configuration::
+
+      -s n,virtio-console,socket:socket_file_name=</path/to/file.sock>:client
+
+#. Login to VM1, check any input to the virtual port(vportnp0, n is decided
+   by your machine)::
+
+      cat /dev/vportnp
```
Comment (ywan170) [inline doc/developer-guides/hld/virtio-console.rst]: should "cat /dev/vportnp0"?

Comment (gaojunhao0504) [reply doc/developer-guides/hld/virtio-console.rst]: I try to use minicom to connect the port, then transport data, then I have modified it, please help to review again, thanks.

File: doc/developer-guides/hld/virtio-console.rst [hunk 3]
```diff
@@ -198,3 +201,35 @@ The File backend only supports console output to a file (no input).
...
+#. Add a pci slot to the device model (``acrn-dm``) command line, adjusting
+   the ``</path/to/file.sock>`` to your use case in the VM2 configuration::
+
+      -s n,virtio-console,socket:socket_file_name=</path/to/file.sock>:client
+
+#. Login to VM1, connect to the virtual port(vport1p0, 1 is decided
+   by front-end driver)::
+
+      minicom -D /dev/vport1p0
+
```
Comment (ywan170) [inline doc/developer-guides/hld/virtio-console.rst]: 1, Please change to #minicom -D /dev/vport1p0.
2, Please change the style to console style, please refer https://projectacrn.github.io/latest/developer-guides/hld/random-virt-hld.html

**Pull request merged**